x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx

```diff
@@ -0,0 +1,260 @@
+                    defaultMessage="Data stream reindex status not available"
+                  />
+                ) : (
+                  <FormattedMessage
+                    id="xpack.upgradeAssistant.index.flyout.detailsStep.reindexingFailedCalloutTitle"
+                    defaultMessage="Data stream reindexing error"
+                  />
+                )
+              }
+            >
+              {reindexState.errorMessage}
+            </EuiCallOut>
+            <EuiSpacer />
+          </>
+        )}
+
+        {reindexState.meta.isFrozen && <FrozenCallOut />}
+        <EuiText>
+          {reindexState.meta.isReadonly && (
+            <Fragment>
+              <p>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.readonlyCompatibleIndexText"
+                  defaultMessage="This index was created in ES 7.x. The index has been flagged as read-only, which allows for N-2 compatibility with the next major version."
+                />
+              </p>
+              <p>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.reindexText"
+                  defaultMessage="The reindex operation allows transforming an index into a new, compatible one. It will copy all of the existing documents into a new index and remove the old one. Depending on size and resources, reindexing may take extended time and your data will be in a read-only state until the job has completed."
+                />
+              </p>
+            </Fragment>
+          )}
+          {!reindexState.meta.isReadonly && (
+            <Fragment>
+              <p>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.notCompatibleIndexText"
+                  defaultMessage="This index was created in ES 7.x and it is not compatible with the next major version. You must address this issue before upgrading."
+                />
+              </p>
+              <ul>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.reindexText"
+                  tagName="li"
+                  defaultMessage="The reindex operation allows transforming an index into a new, compatible one. It will copy all of the existing documents into a new index and remove the old one. Depending on size and resources, reindexing may take extended time and your data will be in a read-only state until the job has completed."
+                />
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.readOnlyText"
+                  tagName="li"
+                  defaultMessage="Alternatively, old indices can maintain compatibility with the next major version if they are turned into read-only mode. If you no longer need to update documents in this index (or add new ones), you might want to convert it to a read-only index."
```

Comment:
I think we need some docs help here: users need to fully understand the implications of addressing incompatibility by making data read-only... I believe there are also search limitations on write blocked data (although I might be wrong).

Either way, I think we need to have a `Learn more` link here.

Reply:
Should we reference https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-blocks.html#index-block-settings or is there a more specific upgrade / read-only doc somewhere?

Reply:
Added __ link referencing the doc proposed above.